src/mbgl/style/layers/layer_properties.hpp.ejs

```diff
@@ -52,6 +59,9 @@ class <%- camelize(type) %>LayoutProperties : public Properties<
 class <%- camelize(type) %>PaintProperties : public Properties<
 <% for (const property of paintProperties.slice(0, -1)) { -%>
     <%- camelize(property.name) %>,
+<% if (property.name === 'line-width') { -%>
+    LineFloorwidth,
```

Comment:
This moves `LineFloorwidth` property up the stack, previously it was scoped within RenderLayer / Bucket (paint prop binder), now it is part of style, thus, used for style diffing, etc. Maybe there is a better solution that allows to keep it within RenderLayer / bucket.

Reply:
At the same time it obviates line layer paint properties concatenation at every frame, which is IMO a heavier operation.